src/address.rs

```diff
@@ -99,35 +99,26 @@ impl Matcher {
         if address.0 == self.pattern {
             return true;
         }
-        let mut remainder: &str = address.0.as_str();
-        // Match the the address component by component
-        for (index, part) in self.pattern_parts.as_slice().iter().enumerate() {
+
+        let mut remainder = address.0.as_str();
+        let mut iter = self.pattern_parts.iter().peekable();
+
+        while !remainder.is_empty() && let Some(part) = iter.next() {
+            // Match the the address component by component
             let result = match part {
-                AddressPatternComponent::Tag(s) => match_literally(remainder, s.as_str()),
+                AddressPatternComponent::Tag(s) => match_literally(remainder, s),
                 AddressPatternComponent::WildcardSingle => match_wildcard_single(remainder),
-                AddressPatternComponent::Wildcard(l) => {
-                    // Check if this is the last pattern component
-                    if index < self.pattern_parts.len() - 1 {
-                        let next = &self.pattern_parts[index + 1];
-                        match next {
-                            // If the next component is a '/', there are no more components in the current part and it can be wholly consumed
-                            AddressPatternComponent::Tag(s) if s == "/" => {
-                                match_wildcard(remainder, *l, None)
-                            }
-                            _ => match_wildcard(remainder, *l, Some(next)),
-                        }
-                    } else {
-                        match_wildcard(remainder, *l, None)
-                    }
-                }
+                AddressPatternComponent::Wildcard(l) => match_wildcard(remainder, *l, iter.peek().copied()),
                 AddressPatternComponent::CharacterClass(cc) => match_character_class(remainder, cc),
                 AddressPatternComponent::Choice(s) => match_choice(remainder, s),
             };
 
-            match result {
-                Ok((i, _)) => remainder = i,
-                Err(_) => return false, // Component didn't match, goodbye
-            };
+            if let Ok((i, _)) = result {
```

Comment:
I'd rather keep the `match` instead of the `if-else` here, but that's personal preference.  Would you be okay with reverting this particular change?

Reply:
Sure, no problem. I’ll do that this afternoon, once I’m back home.

Reply:
Done!